src/Covenant/Data.hs

```diff
@@ -246,12 +251,12 @@ mkBBF (DataDeclaration _ numVars ctors _)
 -- | Packages up all of the relevation datatype information needed
 -- for the ASGBuilder. Note that only certain datatypes have a BB or BB/BF form
 -- (we do not generate forms that are "useless")
-data DatatypeInfo
+data DatatypeInfo (var :: Type)
```

Comment:
Might be worth adding a comment similar to what we do for `ValT` and `CompT` that `var` should only be one of `AbstractTy` or `Renamed`.